### ISSUE TITLE
invalid JPEG format: short segment length

### ISSUE THREAD

**Issue by mike-hart** [imgproxy/imgproxy#203]
Running version 2.4.1 from docker (`docker run -it -p 8080:8080 darthsim/imgproxy:v2.4.1`) and attempting to perform a resize (`resize:fit:800:600:0/gravity:ce`) on the linked image results in:
```
/go/src/github.com/imgproxy/imgproxy/processing_handler.go:123 main.handleProcessing
/go/src/github.com/imgproxy/imgproxy/server.go:81 main.withCORS.func1
/go/src/github.com/imgproxy/imgproxy/router.go:92 main.(*router).ServeHTTP
/usr/lib/go/src/net/http/server.go:2775 net/http.serverHandler.ServeHTTP
/usr/lib/go/src/net/http/server.go:1879 net/http.(*conn).serve
/usr/lib/go/src/runtime/asm_amd64.s:1338 runtime.goexit
```
![a693bf2ab9e14a564ae1269c4e6f7806caf1b09e](https://user-images.githubusercontent.com/4129199/64959727-a90f9280-d889-11e9-83ef-9faee9a751f1.jpg)

**Comment by DarthSim** [comment]
Hi,

Well, this file is actually broken. One of its segments has a negative length, go Go's decoder fails to parse its header: https://github.com/golang/go/blob/master/src/image/jpeg/reader.go#L601-L604. Some apps are more tolerant of this, but Photoshop can't open this file either.

Probably, we should patch the decoder and add it to the repo.

**Comment by mike-hart** [comment]
Fair enough. I meet a few broken images day-to-day (I have Sentry hooked up to imgproxy, so I see them semi-frequently), would you like more submissions like this?

**Comment by DarthSim** [comment]
Yup, that's would be great

**Comment by DarthSim** [comment]
Well, I rewrote all the type & size checking code into the sub-package. It's a lot more tolerant to non-compliance with specifications. Both this image and one from https://github.com/imgproxy/imgproxy/issues/207 work well now.

**Comment by DarthSim** [comment]
Check out the latest build, please.

**Comment by mike-hart** [comment]
Thanks for looking into this. I ran again against latest:

```
$ docker run -it -p 8080:8080 darthsim/imgproxy:latest
2019/09/26 11:23:01 [WARNING] No keys defined, so signature checking is disabled
2019/09/26 11:23:01 [WARNING] No salts defined, so signature checking is disabled
2019/09/26 11:23:01 Starting server at :8080
2019/09/26 11:23:11 [RTt12aI-86QwV1VGOnjo8V] GET: /insecure/resize:fit:800:600:0/gravity:ce/plain/https://user-images.githubusercontent.com/4129199/64959727-a90f9280-d889-11e9-83ef-9faee9a751f1.jpg
2019/09/26 11:23:11 [RTt12aI-86QwV1VGOnjo8V] | 500 | invalid JPEG format: short segment length
/go/src/github.com/imgproxy/imgproxy/processing_handler.go:123 main.handleProcessing
/go/src/github.com/imgproxy/imgproxy/server.go:81 main.withCORS.func1
/go/src/github.com/imgproxy/imgproxy/router.go:92 main.(*router).ServeHTTP
/usr/lib/go/src/net/http/server.go:2775 net/http.serverHandler.ServeHTTP
/usr/lib/go/src/net/http/server.go:1879 net/http.(*conn).serve
/usr/lib/go/src/runtime/asm_amd64.s:1338 runtime.goexit
2019/09/26 11:23:11 [EeP7tt41yHJqdSHQDemjbz] GET: /favicon.ico
```

Unless you mean to build locally from master, or is there a `v2.5.1` due out soon?

**Comment by DarthSim** [comment]
I meant `darthsim/imgproxy:latest`. Maybe you need to re-pull it?

**Comment by mike-hart** [comment]
Yes, sorry, re-pulled `latest` and it's working in test. I look forward to a `v2.5.1`. Thank-you

**Comment by DarthSim** [comment]
Fixed in [2.6.0](https://github.com/imgproxy/imgproxy/releases/tag/v2.6.0)
